migration/migrator/migrations/course/20240324130619_grade_notifications.py

```diff
@@ -0,0 +1,40 @@
+"""Migration for a given Submitty course database."""
+
+
+def up(config, database, semester, course):
+    """
+    Run up migration.
+
+    :param config: Object holding configuration details about Submitty
+    :type config: migrator.config.Config
+    :param database: Object for interacting with given database for environment
+    :type database: migrator.db.Database
+    :param semester: Semester of the course being migrated
+    :type semester: str
+    :param course: Code of course being migrated
+    :type course: str
+    """
+    database.execute(
+        """
+            ALTER TABLE notification_settings ADD COLUMN IF NOT EXISTS all_released_grades BOOLEAN DEFAULT true NOT NULL;
+            ALTER TABLE notification_settings ADD COLUMN IF NOT EXISTS all_released_grades_email BOOLEAN DEFAULT true NOT NULL;
+            ALTER TABLE gradeable ADD COLUMN IF NOT EXISTS g_notification_state BOOLEAN DEFAULT false NOT NULL;            
+            UPDATE gradeable SET g_notification_state = true WHERE g_grade_released_date < NOW();
```

Comment:
I would be worried about a potential (but extremely unlikely) race condition of adding the column and then having the notification cron job get run before the states are able to be marked as false. A possible fix would be to add the column as default true and then mark it false if g_grade_released_date is after now.